twine/repository.py

```diff
@@ -223,7 +223,7 @@ def package_is_uploaded(
                 releases = {}
             self._releases_json_data[safe_name] = releases
 
-        packages = releases.get(package.metadata.version, [])
+        packages = releases.get(cast(str, package.metadata.version), [])
```

Comment:
Same thought about `version` potentially being `None` 🙂

Reply:
Fixed in 4cf7f84.